src/pages/home/report/EmojiPickerMenu/index.js

```diff
@@ -61,6 +90,128 @@ class EmojiPickerMenu extends Component {
         if (this.props.forwardedRef && _.isFunction(this.props.forwardedRef)) {
             this.props.forwardedRef(this.searchInput);
         }
+
+        // Setup keypress/mouse handlers only if we have a keyboard (and not a touchscreen)
+        if (!canUseTouchScreen() && document) {
+            this.keyDownHandler = (e) => {
+                // Highlight the appropriate emoji
+                if (e.key.startsWith('Arrow')) {
+                    this.highlightAdjacentEmoji(e.key);
+                    this.toggleArrowKeysOnSearchInput();
+                }
+
+                // Select the highlighted emoji if enter is pressed
+                if (e.key === 'Enter' && this.state.highlightedIndex !== -1) {
+                    this.props.onEmojiSelected(this.state.filteredEmojis[this.state.highlightedIndex].code);
+                }
+            };
+            document.addEventListener('keydown', this.keyDownHandler);
+
+            // Re-enable pointer events and hovering over EmojiPickerItems when the mouse moves
+            this.mouseMoveHandler = () => {
+                if (this.state.shouldDisablePointerEvents) {
+                    this.setState({shouldDisablePointerEvents: false});
+                }
+            };
+            document.addEventListener('mousemove', this.mouseMoveHandler);
+        }
+    }
+
+    componentWillUnmount() {
+        // Cleanup all mouse/keydown event listeners that we've set up
+        if (!canUseTouchScreen() && document) {
+            document.removeEventListener('keydown', this.keyDownHandler);
+            document.removeEventListener('keydown', this.mouseMoveHandler);
+            this.toggleKeysOnSearchInput([], ['ArrowDown', 'ArrowLeft', 'ArrowRight', 'ArrowUp']);
+        }
+    }
+
+    /**
+     * Highlights emojis adjacent to the currently highlighted emoji depending on the arrowKey
+     * @param {String} arrowKey
+     */
+    highlightAdjacentEmoji(arrowKey) {
+        // If nothing is highlighted and an arrow key is pressed
+        // select the first emoji
+        if (this.state.highlightedIndex === -1) {
+            this.setState(prevState => ({highlightedIndex: prevState.filteredEmojis.length === emojis.length ? 8 : 0}));
+            this.scrollToHighlightedIndex();
+            return;
+        }
+
+        let newIndex = this.state.highlightedIndex;
+        const firstNonHeaderIndex = this.state.filteredEmojis.length === emojis.length ? 8 : 0;
+
+        const move = (steps, boundsCheck) => {
+            if (boundsCheck()) {
+                return;
+            }
+            const isHeader = e => e.header || e.code === CONST.EMOJI_SPACER;
+            do {
+                newIndex += steps;
+            } while (isHeader(this.state.filteredEmojis[newIndex]));
```

Comment:
what is this doing? Not a blocker or anything, but a comment may make things easier to understand for non react native experts.